### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by SarpongAbasimi**
What I did.
- Added a docker file  to dockerize the application
- Added a docker compose file to add `Kakfa`, `zookeeper` and  a `flyway migrator` image
- Modified `DbQueriesSpec.scala` to use `Postgres test containers` 
- Commented out the `web` part of the docker compose file because I am currently facing a config issue. And since it was not part of the original requirement to create a docker image of the whole application, I have decided to look at this later as I think it might take a bit of time to debug.

Issue 

Screenshot 2021-08-04 at 09 00 32

Evidence that Kafka is working.

Screenshot 2021-08-04 at 09 02 51

# Updates Made to the Pr.

- I have changed `user module` to `persistanceService`
- Fixed issue related to db connection
- Added a migration file to create the `dummyProject db`
-  There is still an error when`docker compose up` is run but this is a `good` error. 

Error

Screenshot 2021-08-04 at 15 21 40

The error is due to the fact that the method below has not yet been implemented.

Screenshot 2021-08-04 at 15 23 21

This will be implemented in the next PR when routes are added to the subscription service.

### CODE REVIEW

**Comment by SarpongAbasimi** [comment]
I also just realised that Travis CI is not kicking off so I will also have a look into that.

File: modules/persistenceService/src/main/project/build.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sbt.version=1.5.0
```
Comment (fedefernandez) [inline modules/persistenceService/src/main/project/build.properties]: I think we don't need this

File: modules/persistenceService/src/test/scala/connectionLayer/DbQueriesSpec.scala [hunk 2]
```diff
@@ -4,24 +4,46 @@ import doobie.{Transactor, Update0}
...
+    val driverName    = DriverName(container.driverClassName)
+    val connectionUrl = ConnectionUrl(container.jdbcUrl)
+    val user          = DbUser(container.username)
+    val password      = PassWord(container.password)
 
+    DbMigrations
+      .migrate[IO](ApplicationConfig(driverName, connectionUrl, user, password))
+      .unsafeRunSync()
+    try test()
+    finally {}
```
Comment (fedefernandez) [inline modules/persistenceService/src/test/scala/connectionLayer/DbQueriesSpec.scala]: If we're not using the `finally` we could omit it. Another option would be to encapsulate the `test()` call in an `IO` block and then execute both effects 
```suggestion
    DbMigrations
      .migrate[IO](ApplicationConfig(driverName, connectionUrl, user, password))
      .flatMap(IO(test()))
      .unsafeRunSync()
```

File: modules/persistenceService/src/test/scala/migrations/PsqlMigrationsSpec.scala [hunk 3]
```diff
@@ -22,7 +22,7 @@ class PsqlMigrationsSpec
 
       DbMigrations
         .migrate[IO](ApplicationConfig(driverName, connectionUrl, user, password))
-        .map(_.migrationsExecuted shouldBe (1))
+        .map(_.migrationsExecuted shouldBe (2))
```
Comment (fedefernandez) [inline modules/persistenceService/src/test/scala/migrations/PsqlMigrationsSpec.scala]:
```suggestion
        .map(_.migrationsExecuted > 0)
        .map(_ must beTrue)
```

File: project/build.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+sbt.version = 1.4.7
```
Comment (fedefernandez) [inline project/build.properties]:
```suggestion
sbt.version = 1.5.4
```

**Review by fedefernandez — APPROVED** [review]
Left some minor comments, other than that looks good to me

**Pull request merged**